Feature/cms 660 show read-only settings when `allowAdminChanges` is disabled

i-just opened the pull request:
### Description
Shows read-only settings when `allowAdminChanges` is disabled.

- switched `.revision-notice` to a generic `.content-notice` class so that the styles can be reused for other purposes too
- when admin changes are disallowed, the “Settings” main nav item and all the subpages (except for the plugin settings) are shown, with all the fields and forms being disabled and with a notice at the top of every page
- plugins can opt into having their settings shown when `allowAdminChanges` is disabled; to do so, they need to ensure none of the settings are editable in such case and set `public bool $hasReadOnlyCpSettings` to `true`

Further info are in [Brandon's comment](https://github.com/craftcms/cms/pull/16265#issuecomment-2576459583) below.

<img width="1237" alt="Screenshot 2024-12-06 at 07 47 21" src="https://github.com/user-attachments/assets/09b401a0-d14a-4876-a6ba-110915a16571">



### Related issues
[cms-660](https://linear.app/craftcms/issue/CMS-660/show-read-only-settings-when-allowadminchanges-==-false)

i-just committed:
switch revision notice to a generic class

i-just committed:
show settings > system items in readOnly mode

i-just committed:
no need for a new method...

i-just committed:
show user profile fields but disabled

i-just committed:
show settings > content items in readOnly mode

i-just committed:
show settings > media items in readOnly mode

i-just committed:
handle settings > plugins in readOnly mode

i-just committed:
translation & notice on the main settings page

i-just committed:
build

i-just committed:
missed notices

i-just committed:
Merge remote-tracking branch 'origin/5.6' into feature/cms-660-show-read-only-settings-when-allowadminchanges-false

i-just committed:
email settings > site overrides read only

Mosnar commented:
Love this! It might be worth adding a "Learn More" button to a KB article explaining why settings can't/shouldn't be changed in production. Newer Craft developers or over-confident clients might read the current disclaimer as instructions to simply turn on "allowAdminChanges" and potentially get themselves in a mess.

i-just committed:
twig require admin node and token parser updates

i-just committed:
Merge branch '5.6' into feature/cms-660-show-read-only-settings-when-allowadminchanges-false

i-just committed:
missed from the merge

i-just committed:
updated notice wording

i-just committed:
Merge branch '5.6' into feature/cms-660-show-read-only-settings-when-allowadminchanges-false

gcamacho079 reviewed (changes requested):
@i-just awesome! I pinged you on the Notion page for this feature as I had some follow-up questions.

i-just committed:
add disabled attr to the lightswitch field

i-just committed:
disabled attr for the .menubtn

i-just committed:
adds readOnlySettingsReady method to the transport adapters

i-just committed:
bug fix

i-just committed:
added Field->readOnlySettingsReady()

i-just committed:
disabled field layout designer shouldn't be keyboard focusable

i-just committed:
added Fs->readOnlySettingsReady()

i-just committed:
Merge branch '5.6' into feature/cms-660-show-read-only-settings-when-allowadminchanges-false

i-just commented:
thanks @gcamacho079! I actioned all your comments. Extra info on the Notion page. LMK if anything doesn't make sense.

brandonkelly committed:
Merge branch '5.6' into feature/cms-660-show-read-only-settings-when-allowadminchanges-false

[ci skip]

gcamacho079 commented:
Thanks for knocking out those changes, @i-just! A couple more things I noticed that should also get the `disabled` attribute:

- [x] The `.menubtn` for “Selectable {{ element }} condition” on Entries and Assets field settings pages
- [x] The “Static Rows” field (and the conditionally-revealed fields) on Table field settings
- [x] Info toggle buttons inside disabled table fields (for example, in the “Site Settings” table heading on the Matrix field settings page)

brandonkelly committed:
Merge branch '5.6' into feature/cms-660-show-read-only-settings-when-allowadminchanges-false

[ci skip]

i-just committed:
Merge branch '5.6' into feature/cms-660-show-read-only-settings-when-allowadminchanges-false

i-just committed:
prettier

i-just committed:
added missing disabled attr

i-just committed:
add disabled attr to the settings we can't control

i-just committed:
build

brandonkelly committed:
Merge branch '5.6' into feature/cms-660-show-read-only-settings-when-allowadminchanges-false

brandonkelly committed:
Merge branch '5.6' into feature/cms-660-show-read-only-settings-when-allowadminchanges-false

brandonkelly committed:
Missing allowAdminChanges check

brandonkelly committed:
Be consistent with DdTokenParser etc.

brandonkelly committed:
Reduce allowAdminChangesChecks

- ConfigurableComponent:::getReadOnlySettingsHtml()
- Plugin::getReadOnlySettingsResponse()

brandonkelly committed:
Cp::readOnlyNoticeHtml(), readOnlyNotice()

brandonkelly commented on src/helpers/Cp.php:
```diff
@@ -2917,6 +2930,7 @@ public static function layoutElementSelectorHtml(
                 'has-custom-width' => $element->hasCustomWidth(),
                 'has-settings' => $element->hasSettings(),
             ],
+            'disabled' => true,
```
@i-just What’s the reason for this?

i-just replied:
It was adding a `disabled` attribute to the fields inside the field layout designer’s `.fld-field-library`. But with the changes from https://github.com/craftcms/cms/pull/16374, it’s no longer needed (plus, it looks like I forgot to make it dynamic). I took that out.

brandonkelly committed:
Don't override readOnly

brandonkelly commented:
I [changed things up a bit](https://github.com/craftcms/cms/pull/16265/commits/1a81b86c1a03fabd2ef0870adb15d6581a9f25a3) so now all configurable components have a `getReadOnlySettingsHtml()` method, which by default will return the same thing as `getSettingsHtml()` except with all inputs disabled (similar to `Field::getStaticHtml()`).

Fields, filesystems, and mail transport adapters now override `getReadOnlySettingsHtml()` whenever possible, with the exception of `BaseOptionsField` and `BaseRelationField` (since in those cases, there’s no way to explicitly override `getReadOnlySettingsHtml()` without leaving 3rd party subclasses in the dust, so it’s better to wait until v6 before updating those).

Plugins similarly now have a `getReadOnlySettingsResponse()` method, and a `$hasReadOnlyCpSettings` property. `craft\base\Plugin::getReadOnlySettingsResponse()` will return `$this->getSettingsResponse()` with inputs in the settings HTML disabled (like `ConfigurableComponent::getReadOnlySettingsHtml()` and `Field::getStaticHtml()`), and its `init()` method will automatically set `$this->hasReadOnlyCpSettings = true` if `getSettingsResponse()` hasn’t been overridden.

The point of these changes is to limit the need to check `allowAdminChanges`, in case there’s a reason to render any of these settings without taking `allowAdminChanges` into effect. For example, a form plugin might have a need to render a Plain Text field’s settings even when `allowAdminChanges` is disabled (because maybe the field isn’t getting saved to the project config), and should still be able to call `$field->getSettingsHtml()` and get the normal non-disabled response.

i-just committed:
move noroutes container into routes so that readOnlyNotice shows correctly

i-just committed:
add missing since tags

i-just committed:
remove disabled attribute from layoutElementSelectorHtml

i-just committed:
link type settings should also respect $readOnly param

brandonkelly committed:
Call getReadOnlySettingsHtml() on the link types

brandonkelly committed:
Better Html::disableInputs() implementation using DomCrawler

brandonkelly committed:
Allow a callable to be passed into Html::disableInputs()

brandonkelly committed:
Show a preview input for autosuggest fields

Otherwise, w/o any JS registered, browsers don't know what to do with the <vue-autosuggest> tag.

brandonkelly committed:
Simpler

brandonkelly committed:
Release notes

[ci skip]
